From 11c858963df7d68c2b381f1dab8547a9ca7947d4 Mon Sep 17 00:00:00 2001
From: "Reddy, Alavala Srinivasa" <alavala.srinivasa.reddy@intel.com>
Date: Thu, 2 Feb 2023 14:44:56 +0530
Subject: [PATCH] Fix for ASB-Feb-2023 Build issue

Do not clear calling identify when using
BiometricPrompt from FingerprintService.

Bug: 214261879
Test: atest AuthControllerTest
Test: Manually verify with test apps in bug

Signed-off-by: Reddy, Alavala Srinivasa <alavala.srinivasa.reddy@intel.com>
---
 core/java/android/hardware/biometrics/BiometricPrompt.java      | 2 +-
 .../biometrics/sensors/fingerprint/FingerprintService.java      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/java/android/hardware/biometrics/BiometricPrompt.java b/core/java/android/hardware/biometrics/BiometricPrompt.java
index e7d7c3cdf528..a54c26d72eae 100644
--- a/core/java/android/hardware/biometrics/BiometricPrompt.java
+++ b/core/java/android/hardware/biometrics/BiometricPrompt.java
@@ -899,7 +899,7 @@ public class BiometricPrompt implements BiometricAuthenticator, BiometricConstan
      * @hide
      */
     @RequiresPermission(USE_BIOMETRIC)
-    public void authenticateForOperation(
+    public long authenticateForOperation(
             @NonNull CancellationSignal cancel,
             @NonNull @CallbackExecutor Executor executor,
             @NonNull AuthenticationCallback callback,
diff --git a/services/core/java/com/android/server/biometrics/sensors/fingerprint/FingerprintService.java b/services/core/java/com/android/server/biometrics/sensors/fingerprint/FingerprintService.java
index bd42a17d5ec2..6c12e93cded4 100644
--- a/services/core/java/com/android/server/biometrics/sensors/fingerprint/FingerprintService.java
+++ b/services/core/java/com/android/server/biometrics/sensors/fingerprint/FingerprintService.java
@@ -422,7 +422,7 @@ public class FingerprintService extends SystemService {
                         }
                     };
 
-            return biometricPrompt.authenticateUserForOperation(
+            return biometricPrompt.authenticateForOperation(
                     new CancellationSignal(), executor, promptCallback, operationId);
         }
 
-- 
2.17.1

